src/threepp/renderers/gl/GLObjects.cpp

```diff
@@ -67,9 +49,17 @@ struct GLObjects::Impl {
 
         if (auto instancedMesh = object->as<InstancedMesh>()) {
 
-            if (!object->hasEventListener("dispose", &onInstancedMeshDispose)) {
+            if (instanceMeshDisposeSubscriptions_.find(object) == instanceMeshDisposeSubscriptions_.end()) {
+
+                auto onInstanceMeshDispose = [this,object](Event& event) {
+                    auto instancedMesh = static_cast<InstancedMesh*>(event.target);
+                    this->attributes_.remove(instancedMesh->instanceMatrix());
+                    if (instancedMesh->instanceColor()) this->attributes_.remove(instancedMesh->instanceColor());
+                    // Remove our subscription
+                    this->instanceMeshDisposeSubscriptions_.erase(object);
```

Comment:
Invokes UB as EventListener::send is currently looping over `listeners_` and the custom deleter modifies it.

Reply:
That's embarrassing 🤦. I fixed this in #239